source/Weather.h

```diff
@@ -50,7 +51,7 @@ class Weather {
 	// The origin of the hazard.
 	const Point &Origin() const;
 	// Create any environmental effects and decrease the lifetime of this weather.
-	void Step(std::vector<Visual> &newVisuals);
+	void Step(std::vector<Visual> &newVisuals, const Point *effectCenter = nullptr);
```

Comment:
It indeed does, that is just the position of the flagship, if available and then the effect decides if it should use it.
Maybe I should rename it to be less vague?

Reply:
This code is okay if the hazardCenter and the effectCenter are at the same location. I somehow had the impression that only the effectCenter was moved and the hazardCenter was not, but you can ignore my remark here if both centers are on the same location.

Reply:
The effectCenter* is only used when the hazard is system-wide, to draw it only around the flagship of the player. Given that it is system-wide, there is no point moving the hazard itself, it should always originate from its centre for the calculations (if its an explosion), whether that be the system centre or a planet.

Reply:
Sorry, I was not communicating very clearly; I was thinking of a new type of "system wide effect" that could be used on hazards and on weapons. (We can then even create weapons that generate "system wide effects").

The conceptual difference is that this PR disconnects the center of the hazard from the center of the effect, while the "system wide effect" just doesn't have any center; it just appears system-wide (and only gets generated in the visible viewport).

Reply:
So you want system-wide effect to be separate from system-wide hazard, so if say smbd forgets to include it into the effect as well for a system-wide hazard, what happens? That's my issue, you can't just have the hazard modify its effect, and we can't assume nobody forgets it? (I dont think tehhowch would allow that)

On top of that, system-wide effect on a weapon is smth that didn't occur to me I must say, no idea why anyone would want that and that was the main argument for not doing the hazard following the flagship, usability? If we were to do smth like that we would expect the weapon to have an actual effect on the entire system, or imagine a weapon hitting one of your ships and all of a sudden you see particle but that's all, makes no sense.

I do see one possibility to do what you are saying, and it is to move the range of the hazard into the effect of said hazard, which would break all existing hazards and is thus a no-go too.

Reply:
Maybe we should redefine what amount means overall? Say have it be by 1000x1000 pixels, always?

Reply:
> Maybe we should redefine what amount means overall? Say have it be by 1000x1000 pixels, always?

Sounds good.